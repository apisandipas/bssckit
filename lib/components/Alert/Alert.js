"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.AlertDismissIcon = void 0;

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _base = require("../Utilities/base");

var _themeFunctions = require("../../utils/themeFunctions");

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _templateObject11() {
  var data = _taggedTemplateLiteral(["\n  position: relative;\n  padding: ", ";\n  border: ", ";\n  & > a {\n    font-weight: ", ";\n  };\n  & > h1,\n  & > h2,\n  & > h3,\n  & > h4,\n  & > h5,\n  & > h6 {\n    color: inherit;\n  };\n  ", "\n  ", "\n  ", "\n  ", "\n  ", "\n  ", "\n  ", "\n\n"]);

  _templateObject11 = function _templateObject11() {
    return data;
  };

  return data;
}

function _templateObject10() {
  var data = _taggedTemplateLiteral(["\n  font-size: 2rem;\n  cursor: pointer;\n  line-height: 1rem;\n  position: absolute;\n  top: 1rem;\n  right: 1rem;\n  color: inherit;\n  opacity: 0.5;\n\n  &:hover,\n  &:focus {\n    opacity: 0.75;\n  }\n"]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = _taggedTemplateLiteral(["\n    border-radius: ", ";\n  "]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _taggedTemplateLiteral(["\n      border-radius: ", ";\n    "]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _taggedTemplateLiteral(["\n      border-radius: ", ";\n    "]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteral(["\n    & > a,\n    & > a:hover {\n      color: ", ";\n    }\n  "]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteral(["\n    & > hr {\n      border-top-color: ", ";\n    }\n  "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteral(["\n    border-color: ", ";\n  "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteral(["\n    background-color: ", ";\n  "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["\n    color: ", ";\n  "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n    padding-right: ", ";\n  "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var alertDismissible = function alertDismissible(props) {
  return props.dismissible && (0, _styledComponents.css)(_templateObject(), (0, _themeFunctions.getPadding)(props, 'alert', 'dismissibleRight'));
};

var color = function color(props) {
  return (0, _styledComponents.css)(_templateObject2(), (0, _themeFunctions.getColor)(props, 'alert', 'text'));
};

var backgroundColor = function backgroundColor(props) {
  return (0, _styledComponents.css)(_templateObject3(), (0, _themeFunctions.getColor)(props, 'alert', 'background'));
};

var borderColor = function borderColor(props) {
  return (0, _styledComponents.css)(_templateObject4(), (0, _themeFunctions.getColor)(props, 'alert', 'border'));
};

var borderTopColor = function borderTopColor(props) {
  return (0, _styledComponents.css)(_templateObject5(), (0, _themeFunctions.getColor)(props, 'alert', 'borderTop'));
};

var linkHoverColor = function linkHoverColor(props) {
  return (0, _styledComponents.css)(_templateObject6(), (0, _themeFunctions.getColor)(props, 'alert', 'hover'));
};

var borderRadius = function borderRadius(props) {
  if (props.noRadius) {
    return (0, _styledComponents.css)(_templateObject7(), (0, _themeFunctions.getBorderRadius)(props, 'alert', 'noRadius'));
  } else if (props.pill) {
    return (0, _styledComponents.css)(_templateObject8(), (0, _themeFunctions.getBorderRadius)(props, 'alert', 'pill'));
  }

  return (0, _styledComponents.css)(_templateObject9(), (0, _themeFunctions.getBorderRadius)(props, 'alert', 'default'));
};

var AlertDismissIcon = _styledComponents["default"].span.attrs({
  ariaHidden: 'true'
})(_templateObject10());

exports.AlertDismissIcon = AlertDismissIcon;
var Alert = (0, _styledComponents["default"])(_base.Div).attrs({
  role: 'alert'
})(_templateObject11(), function (props) {
  return (0, _themeFunctions.getPadding)(props, 'alert', 'default');
}, function (props) {
  return (0, _themeFunctions.getBorder)(props, 'alert', 'default');
}, function (props) {
  return (0, _themeFunctions.getFontWeight)(props, 'alert', 'default');
}, function (props) {
  return borderRadius(props);
}, function (props) {
  return alertDismissible(props);
}, function (props) {
  return color(props);
}, function (props) {
  return backgroundColor(props);
}, function (props) {
  return borderColor(props);
}, function (props) {
  return borderTopColor(props);
}, function (props) {
  return linkHoverColor(props);
});
var _default = Alert;
exports["default"] = _default;