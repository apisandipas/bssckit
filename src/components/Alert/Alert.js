import styled, { css } from 'styled-components';
import { Div } from '../Utilities/base';
import { getColor, getPadding, getBorder, getFontWeight, getBorderRadius } from '../../utils/themeFunctions';

const alertDismissible = props =>
  props.dismissible &&
  css`
    padding-right: ${getPadding(props, 'alert', 'dismissibleRight')};
  `;

const color = props =>
  css`
    color: ${getColor(props, 'alert', 'text')};
  `;

const backgroundColor = props =>
  css`
    background-color: ${getColor(props, 'alert', 'background')};
  `;

const borderColor = props =>
  css`
    border-color: ${getColor(props, 'alert', 'border')};
  `;

const borderTopColor = props =>
  css`
    & > hr {
      border-top-color: ${getColor(props, 'alert', 'borderTop')};
    }
  `;

const linkHoverColor = props =>
  css`
    & > a,
    & > a:hover {
      color: ${getColor(props, 'alert', 'hover')};
    }
  `;

const borderRadius = props => {
  if (props.noRadius) {
    return css`
      border-radius: ${getBorderRadius(props, 'alert', 'noRadius')};
    `;
  } else if (props.pill) {
    return css`
      border-radius: ${getBorderRadius(props, 'alert', 'pill')};
    `;
  }

  return css`
    border-radius: ${getBorderRadius(props, 'alert', 'default')};
  `;
};

export const AlertDismissIcon = styled.span.attrs({
  ariaHidden: 'true',
})`
  font-size: 2rem;
  cursor: pointer;
  line-height: 1rem;
  position: absolute;
  top: 1rem;
  right: 1rem;
  color: inherit;
  opacity: 0.5;

  &:hover,
  &:focus {
    opacity: 0.75;
  }
`;

const Alert = styled(Div).attrs({
  role: 'alert',
})`
  position: relative;
  padding: ${props => getPadding(props, 'alert', 'default')};
  border: ${props => getBorder(props, 'alert', 'default')};
  & > a {
    font-weight: ${props => getFontWeight(props, 'alert', 'default')};
  };
  & > h1,
  & > h2,
  & > h3,
  & > h4,
  & > h5,
  & > h6 {
    color: inherit;
  };
  ${props => borderRadius(props)}
  ${props => alertDismissible(props)}
  ${props => color(props)}
  ${props => backgroundColor(props)}
  ${props => borderColor(props)}
  ${props => borderTopColor(props)}
  ${props => linkHoverColor(props)}

`;

export default Alert;
