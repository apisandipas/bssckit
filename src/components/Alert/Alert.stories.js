import React, { useState } from 'react';
import { Alert, AlertDismissIcon, DismissableAlert } from './';
import { Story } from '../../utils/functions';
import BSThemeProvider from '../Utilities/BSThemeProvider';

export default {
  title: 'Alert',
};

export const simple = () => <Alert primary>Hello World</Alert>;

export const noRadius = () => (
  <Alert secondary noRadius>
    Hello World
  </Alert>
);

export const pill = () => (
  <Alert info pill>
    Hello World
  </Alert>
);

export const contextual = () => (
  <>
    <Alert primary mb4>
      Primary alert
    </Alert>
    <Alert secondary mb4>
      Secondary alert
    </Alert>
    <Alert success mb4>
      Success alert
    </Alert>
    <Alert danger mb4>
      Danger alert
    </Alert>
    <Alert warning mb4>
      Warning alert
    </Alert>
    <Alert info mb4>
      Info alert
    </Alert>
    <Alert light mb4>
      Light alert
    </Alert>
    <Alert dark>Dark alert</Alert>
  </>
);

export const withALink = () => (
  <>
    <Alert primary mb4>
      This is a primary alert with <a href='#'>an example link</a>. Give it a click if you like.
    </Alert>
    <Alert danger>
      This is a primary alert with <a href='#'>an example link</a>. Give it a click if you like.
    </Alert>
  </>
);

export const staticDismissible = () => (
  <>
    <Alert primary dismissible>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Iste tempore ea distinctio iure quas quibusdam
      dolorem inventore dolorum explicabo a aspernatur quisquam tempora, sapiente sed repudiandae, iusto
      architecto ipsum blanditiis.
      <AlertDismissIcon>&times;</AlertDismissIcon>
    </Alert>
  </>
);

export const dismissible = Story(() => {
  const [hidden, setHidden] = useState(false);
  return (
    <DismissableAlert danger toggle={() => setHidden(!hidden)} isOpen={!hidden}>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Iste tempore ea distinctio iure quas quibusdam
      dolorem inventore dolorum explicabo a aspernatur quisquam tempora, sapiente sed repudiandae, iusto
      architecto ipsum blanditiis.
    </DismissableAlert>
  );
});

const theme = {
  alert: {
    colors: {
      default: {
        text: 'white',
        background: 'cornflowerblue',
      },
    },
  },
};

export const withTheme = () => (
  <BSThemeProvider theme={theme}>
    <Alert>Always blue! Always blue!</Alert>
  </BSThemeProvider>
);

const theme2 = {
  palette: {
    primaryLight: '#bada55',
    dangerLight: '#db84be',
  },
};

export const withPaletteOverride = () => (
  <BSThemeProvider theme={theme2}>
    <Alert primary mb4>
      Testing 123!
    </Alert>
    <Alert danger>Testing 123!</Alert>
  </BSThemeProvider>
);
