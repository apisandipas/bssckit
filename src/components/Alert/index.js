import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import { theme, ifProp } from 'styled-tools';
import { Div } from '../base';
export { default as DismissableAlert } from './DismissableAlert';
import defaultTheme from '../../utils/defaultTheme';
import { themeContexts } from '../../utils/consts';

const { alert } = defaultTheme;

export const AlertDismissIcon = styled.span.attrs({
  ariaHidden: 'true',
})`
  font-size: 2rem;
  cursor: pointer;
  line-height: 1rem;
  position: absolute;
  top: 1rem;
  right: 1rem;
  color: inherit;
  opacity: 0.5;

  &:hover,
  &:focus {
    opacity: 0.75;
  }
`;

const Alert = styled(Div).attrs({
  role: 'alert',
})`
  position: relative;
  padding: ${theme('alert.padding.default', alert.padding.default)};
  margin-bottom: ${theme('alert.margin.bottom', alert.margin.bottom)};
  border: ${theme('alert.border', alert.border.default)};
  border-radius: ${theme('alert.borderRadius.default', alert.borderRadius.default)};

  & > a,
  & > a:hover {
    font-weight: ${theme('alert.fontWeight.default', alert.fontWeight.default)};
  }

  & > h1,
  & > h2,
  & > h3,
  & > h4,
  & > h5,
  & > h6 {
    color: inherit;
  }

  ${themeContexts.map(context => {
    const { text, background, border } = defaultTheme.alert.colors[context];

    return ifProp(
      context,
      css`
        color: ${theme(`colorScheme.${context}Darkest`, text)};
        background-color: ${theme(`colorScheme.${context}Light`, background)};
        border-color: ${theme(`colorScheme.${context}Lighter`, border)};

        & > a,
        & > a:hover {
          color: ${theme(`colorScheme.${context}Darkest`, text)};
        }

        & > h1,
        & > h2,
        & > h3,
        & > h4,
        & > h5,
        & > h6 {
          color: ${theme(`colorScheme.${context}Darkest`, text)};
        }

        & > hr {
          border-top-color: ${theme(`colorScheme.${context}Darkest`, text)};
        }
      `
    );
  })}

  ${ifProp(
    'noRadius',
    css`
      border-radius: 0;
    `
  )}

  ${ifProp(
    'pill',
    css`
      border-radius: 10rem;
    `
  )}

  ${ifProp(
    'dismissible',
    css`
      padding-right: ${theme('alert.padding.dismissibleRight', alert.padding.dismissibleRight)};
    `
  )};
`;

export default Alert;
