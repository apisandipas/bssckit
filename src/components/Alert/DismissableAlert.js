import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Alert, { AlertDismissIcon } from './Alert';
import { tagPropType } from '../../utils/consts';
import Fade from '../../utils/Fade';

const DismissButton = styled.button`
  padding: 0;
  background-color: transparent;
  border: 0;
  appearance: none;
`;

const DismissableAlert = props => {
  const {
    className,
    closeClassName,
    closeAriaLabel,
    tag: Tag,
    isOpen,
    toggle,
    children,
    transition,
    fade,
    innerRef,
    ...attributes
  } = props;

  const alertTransition = {
    ...Fade.defaultProps,
    ...transition,
    baseClass: fade ? transition.baseClass : '',
    timeout: fade ? transition.timeout : 0,
  };

  return (
    <Fade {...attributes} {...alertTransition} tag={Alert} in={isOpen} role='alert' innerRef={innerRef}>
      {children}
      {toggle ? (
        <DismissButton type='button' aria-label={closeAriaLabel} onClick={toggle}>
          <AlertDismissIcon aria-hidden='true' aria-label={closeAriaLabel} onClick={toggle}>
            &times;
          </AlertDismissIcon>
        </DismissButton>
      ) : null}
    </Fade>
  );
};

DismissableAlert.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  closeClassName: PropTypes.string,
  closeAriaLabel: PropTypes.string,
  color: PropTypes.string,
  fade: PropTypes.bool,
  isOpen: PropTypes.bool,
  toggle: PropTypes.func,
  tag: tagPropType,
  transition: PropTypes.shape(Fade.propTypes),
  innerRef: PropTypes.oneOfType([PropTypes.object, PropTypes.string, PropTypes.func]),
};

DismissableAlert.defaultProps = {
  dismissible: true,
  isOpen: true,
  tag: Alert,
  closeAriaLabel: 'Close',
  fade: true,
  transition: {
    ...Fade.defaultProps,
    unmountOnExit: true,
  },
};

export default DismissableAlert;
