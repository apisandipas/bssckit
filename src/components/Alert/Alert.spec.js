import React from 'react';
import { axe } from 'jest-axe';
import { render } from '@testing-library/react';
import 'jest-styled-components';
import Alert from './Alert';

describe('Alert Component', () => {
  it('matches plain', async () => {
    const { container } = render(<Alert>Hello World</Alert>);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyleRule('position', 'relative');
    expect(container.firstChild).toHaveStyleRule('padding', '0.75rem 1.25rem');
    expect(container.firstChild).toHaveStyleRule('border', '1px solid transparent');
    expect(container.firstChild).toHaveStyleRule('border-radius', '0.25rem');
    expect(container.firstChild).toHaveStyleRule('font-weight', '700', {
      modifier: '& > a',
    });
    expect(container.firstChild).toHaveStyleRule('color', 'inherit', {
      modifier: '& > h1',
    });
    expect(container.firstChild).toHaveStyleRule('color', 'inherit', {
      modifier: '& > h2',
    });
    expect(container.firstChild).toHaveStyleRule('color', 'inherit', {
      modifier: '& > h3',
    });
    expect(container.firstChild).toHaveStyleRule('color', 'inherit', {
      modifier: '& > h4',
    });
    expect(container.firstChild).toHaveStyleRule('color', 'inherit', {
      modifier: '& > h5',
    });
    expect(container.firstChild).toHaveStyleRule('color', 'inherit', {
      modifier: '& > h6',
    });
  });

  it('matches primary', async () => {
    const { container } = render(<Alert primary />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches secondary', async () => {
    const { container } = render(<Alert secondary />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches success', async () => {
    const { container } = render(<Alert success />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches danger', async () => {
    const { container } = render(<Alert danger />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches warning', async () => {
    const { container } = render(<Alert warning />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches info', async () => {
    const { container } = render(<Alert info />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches light', async () => {
    const { container } = render(<Alert light />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches dark', async () => {
    const { container } = render(<Alert dark />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches noRadius', async () => {
    const { container } = render(<Alert noRadius />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyleRule('border-radius', '0');
  });

  it('matches pill', async () => {
    const { container } = render(<Alert pill />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyleRule('border-radius', '10rem');
  });
});
