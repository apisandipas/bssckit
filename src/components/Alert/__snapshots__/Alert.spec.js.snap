// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Component matches danger 1`] = `
.c0 {
  position: relative;
  padding: 0.75rem 1.25rem;
  border: 1px solid transparent;
  border-radius: 0.25rem;
  color: #721c24;
  background-color: #f8d7da;
  border-color: #f5b4bb;
}

.c0 > a {
  font-weight: 700;
}

.c0 > h1,
.c0 > h2,
.c0 > h3,
.c0 > h4,
.c0 > h5,
.c0 > h6 {
  color: inherit;
}

.c0 > hr {
  border-top-color: #f5b4bb;
}

.c0 > a,
.c0 > a:hover {
  color: #721c24;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<div
  class="sc-fznyAO c0"
  role="alert"
/>
`;

exports[`Alert Component matches dark 1`] = `
.c0 {
  position: relative;
  padding: 0.75rem 1.25rem;
  border: 1px solid transparent;
  border-radius: 0.25rem;
  color: #171a1d;
  background-color: #d6d8d9;
  border-color: #abaeb1;
}

.c0 > a {
  font-weight: 700;
}

.c0 > h1,
.c0 > h2,
.c0 > h3,
.c0 > h4,
.c0 > h5,
.c0 > h6 {
  color: inherit;
}

.c0 > hr {
  border-top-color: #abaeb1;
}

.c0 > a,
.c0 > a:hover {
  color: #171a1d;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<div
  class="sc-fznyAO c0"
  role="alert"
/>
`;

exports[`Alert Component matches info 1`] = `
.c0 {
  position: relative;
  padding: 0.75rem 1.25rem;
  border: 1px solid transparent;
  border-radius: 0.25rem;
  color: #10707f;
  background-color: #d1ecf1;
  border-color: #a3d6de;
}

.c0 > a {
  font-weight: 700;
}

.c0 > h1,
.c0 > h2,
.c0 > h3,
.c0 > h4,
.c0 > h5,
.c0 > h6 {
  color: inherit;
}

.c0 > hr {
  border-top-color: #a3d6de;
}

.c0 > a,
.c0 > a:hover {
  color: #10707f;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<div
  class="sc-fznyAO c0"
  role="alert"
/>
`;

exports[`Alert Component matches light 1`] = `
.c0 {
  position: relative;
  padding: 0.75rem 1.25rem;
  border: 1px solid transparent;
  border-radius: 0.25rem;
  background-color: #fefefe;
  border-color: #cccecf;
}

.c0 > a {
  font-weight: 700;
}

.c0 > h1,
.c0 > h2,
.c0 > h3,
.c0 > h4,
.c0 > h5,
.c0 > h6 {
  color: inherit;
}

.c0 > hr {
  border-top-color: #fdfdfe;
}

.c0 > a,
.c0 > a:hover {
  color: #171a1d;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<div
  class="sc-fznyAO c0"
  role="alert"
/>
`;

exports[`Alert Component matches noRadius 1`] = `
.c0 {
  position: relative;
  padding: 0.75rem 1.25rem;
  border: 1px solid transparent;
  border-radius: 0;
  color: #383d41;
  background-color: #e2e3e5;
  border-color: #b4b5b7;
}

.c0 > a {
  font-weight: 700;
}

.c0 > h1,
.c0 > h2,
.c0 > h3,
.c0 > h4,
.c0 > h5,
.c0 > h6 {
  color: inherit;
}

.c0 > hr {
  border-top-color: #b4b5b7;
}

.c0 > a,
.c0 > a:hover {
  color: #383d41;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<div
  class="sc-fznyAO c0"
  role="alert"
/>
`;

exports[`Alert Component matches pill 1`] = `
.c0 {
  position: relative;
  padding: 0.75rem 1.25rem;
  border: 1px solid transparent;
  border-radius: 10rem;
  color: #383d41;
  background-color: #e2e3e5;
  border-color: #b4b5b7;
}

.c0 > a {
  font-weight: 700;
}

.c0 > h1,
.c0 > h2,
.c0 > h3,
.c0 > h4,
.c0 > h5,
.c0 > h6 {
  color: inherit;
}

.c0 > hr {
  border-top-color: #b4b5b7;
}

.c0 > a,
.c0 > a:hover {
  color: #383d41;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<div
  class="sc-fznyAO c0"
  role="alert"
/>
`;

exports[`Alert Component matches plain 1`] = `
.c0 {
  position: relative;
  padding: 0.75rem 1.25rem;
  border: 1px solid transparent;
  border-radius: 0.25rem;
  color: #383d41;
  background-color: #e2e3e5;
  border-color: #b4b5b7;
}

.c0 > a {
  font-weight: 700;
}

.c0 > h1,
.c0 > h2,
.c0 > h3,
.c0 > h4,
.c0 > h5,
.c0 > h6 {
  color: inherit;
}

.c0 > hr {
  border-top-color: #b4b5b7;
}

.c0 > a,
.c0 > a:hover {
  color: #383d41;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<div
  class="sc-fznyAO c0"
  role="alert"
>
  Hello World
</div>
`;

exports[`Alert Component matches primary 1`] = `
.c0 {
  position: relative;
  padding: 0.75rem 1.25rem;
  border: 1px solid transparent;
  border-radius: 0.25rem;
  color: #004085;
  background-color: #cce5ff;
  border-color: #9cc9fb;
}

.c0 > a {
  font-weight: 700;
}

.c0 > h1,
.c0 > h2,
.c0 > h3,
.c0 > h4,
.c0 > h5,
.c0 > h6 {
  color: inherit;
}

.c0 > hr {
  border-top-color: #9cc9fb;
}

.c0 > a,
.c0 > a:hover {
  color: #004085;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<div
  class="sc-fznyAO c0"
  role="alert"
/>
`;

exports[`Alert Component matches secondary 1`] = `
.c0 {
  position: relative;
  padding: 0.75rem 1.25rem;
  border: 1px solid transparent;
  border-radius: 0.25rem;
  color: #383d41;
  background-color: #e2e3e5;
  border-color: #b4b5b7;
}

.c0 > a {
  font-weight: 700;
}

.c0 > h1,
.c0 > h2,
.c0 > h3,
.c0 > h4,
.c0 > h5,
.c0 > h6 {
  color: inherit;
}

.c0 > hr {
  border-top-color: #b4b5b7;
}

.c0 > a,
.c0 > a:hover {
  color: #383d41;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<div
  class="sc-fznyAO c0"
  role="alert"
/>
`;

exports[`Alert Component matches success 1`] = `
.c0 {
  position: relative;
  padding: 0.75rem 1.25rem;
  border: 1px solid transparent;
  border-radius: 0.25rem;
  color: #155724;
  background-color: #d4edda;
  border-color: #72d889;
}

.c0 > a {
  font-weight: 700;
}

.c0 > h1,
.c0 > h2,
.c0 > h3,
.c0 > h4,
.c0 > h5,
.c0 > h6 {
  color: inherit;
}

.c0 > hr {
  border-top-color: #72d889;
}

.c0 > a,
.c0 > a:hover {
  color: #155724;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<div
  class="sc-fznyAO c0"
  role="alert"
/>
`;

exports[`Alert Component matches warning 1`] = `
.c0 {
  position: relative;
  padding: 0.75rem 1.25rem;
  border: 1px solid transparent;
  border-radius: 0.25rem;
  color: #856404;
  background-color: #fff3cd;
  border-color: #f3da8d;
}

.c0 > a {
  font-weight: 700;
}

.c0 > h1,
.c0 > h2,
.c0 > h3,
.c0 > h4,
.c0 > h5,
.c0 > h6 {
  color: inherit;
}

.c0 > hr {
  border-top-color: #f3da8d;
}

.c0 > a,
.c0 > a:hover {
  color: #856404;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<div
  class="sc-fznyAO c0"
  role="alert"
/>
`;
